staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/apiserver.go

```diff
@@ -210,6 +211,18 @@ func (c completedConfig) New(delegationTarget genericapiserver.DelegationTarget)
 		return nil
 	})
 
+	// we don't want to report healthy until we can handle all CRDs that have already been registered.  Waiting for the informer
+	// to sync makes sure that the lister will be valid before we begin.  There may still be races for CRDs added after startup,
```

Comment:
This is true for readiness check, but not for liveness check. If we aren't healthy, kubelet will restart us. Is that what we want?

Reply:
> This is true for readiness check, but not for liveness check. If we aren't healthy, kubelet will restart us. Is that what we want?

`HasSynced` is a "past the post" function that never returns to a false state.